### PR TITLE
🎨 (frontend) implement cunningham Button

### PR DESCRIPTION
**Pull request by rlecellier**
## Purpose

Use cunningham design system for our buttons

All commit will be squashed before merge.

Here the cmd i used to find button to switch to richie : 
```sh
git grep '<button'  | grep -v spec | grep -v stories | grep -v __mocks__ | grep 'button'
```

### CODE REVIEW

File: src/frontend/js/widgets/Search/components/SearchFiltersPane/index.tsx [hunk 1]
```diff
@@ -74,10 +75,13 @@ export const SearchFiltersPane = ({
       >
         <FormattedMessage {...messages.filter} />
       </h2>
-      <button
-        className={`search-filters-pane__clear ${
-          !activeFilterCount ? 'search-filters-pane__clear--hidden' : ''
-        }`}
+      <Button
+        // color="tertiary"
```
Comment (jbpenrath) [inline src/frontend/js/widgets/Search/components/SearchFiltersPane/index.tsx:79]: Debug residue ?

File: src/frontend/js/widgets/UserLogin/index.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      source: {
+        code: 'Disabled for this story, see https://github.com/storybookjs/storybook/issues/11554',
+      },
+    },
+  },
+  args: {
+    context: RichieContextFactory().one(),
+    profileUrls: {
+      first_entry: { action: () => {}, label: 'First entry' },
+      second_entry: { action: () => {}, label: 'Sedonc entry' },
```
Comment (jbpenrath) [inline src/frontend/js/widgets/UserLogin/index.stories.tsx:19]: Qu'sedonc ? 😛

```suggestion
      second_entry: { action: () => {}, label: 'Second entry' },
```

File: src/frontend/js/components/AddressesManagement/index.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+      },
+    },
+  },
+  args: {
+    context: RichieContextFactory().one(),
+    profileUrls: {
+      first_entry: { action: () => {}, label: 'First entry' },
+      second_entry: { action: () => {}, label: 'Sedonc entry' },
+    },
+  },
```
Comment (jbpenrath) [inline src/frontend/js/components/AddressesManagement/index.stories.tsx:21]: Copy pasta failed ?

File: src/frontend/js/components/Modal/_styles.scss [hunk 4]
```diff
@@ -31,20 +31,12 @@
   }
 
   &__closeButton {
-    @include button-reset-style();
-    background-color: r-theme-val(modal, close-background);
-    @if r-theme-val(modal, close-border) != null {
-      border: thin solid r-theme-val(modal, close-border);
-    }
     box-sizing: initial;
-    color: r-theme-val(modal, close-color);
```
Comment (jbpenrath) [inline src/frontend/js/components/Modal/_styles.scss:40]: If we remove `close-*` variables from `modal` theme, we should remove them from `_theme.scss` then add an entry into the `UPGRADE.md`.

File: src/frontend/js/widgets/Dashboard/components/DashboardCard/index.tsx [hunk 5]
```diff
@@ -1,6 +1,6 @@
 import { PropsWithChildren, ReactNode, useMemo, useRef, useState } from 'react';
```
Comment (jbpenrath) [inline src/frontend/js/widgets/Dashboard/components/DashboardCard/index.tsx:1]: In the story of this component, there are remaining Button components to replace.

**Review by jbpenrath — COMMENTED** [review]
Great work ! 

There are a lot of commits but it really helps to focus on each update for the review.
In the other hand, you will have to add a commit message to each commit... or maybe at the end of the review we could squash all commits in two ? (One for migration, one for stories) WDYT ?

File: src/frontend/js/widgets/CourseProductItem/components/PaymentButton/index.tsx [hunk 6]
```diff
@@ -224,8 +225,7 @@ const PaymentButton = ({ product, billingAddress, creditCard, onSuccess }: Payme
 
   return (
     <div className="payment-button">
-      <button
```
Comment (jbpenrath) [inline src/frontend/js/widgets/CourseProductItem/components/PaymentButton/index.tsx:227]: Sale buttons are currently displayed with Hind typography but before there were displayed with Montserrat. WDYT to customize cunningham button font family to Montserrat ?

File: src/frontend/js/components/RegisteredAddress/index.tsx [hunk 7]
```diff
@@ -86,30 +82,32 @@ const RegisteredAddress = ({ promote, select, edit, remove, address }: Props) =>
...
         >
           <FormattedMessage {...messages.editButton} />
-        </button>
-        <button
+        </Button>
+        <Button
           aria-label={intl.formatMessage(messages.deleteButtonLabel, {
             title: address.title,
           })}
-          className="button button--tiny button--pill  button-sale--secondary"
```
[the remark this answers is not in the document]
Comment (rlecellier) [reply src/frontend/js/components/RegisteredAddress/index.tsx:107]: There a visibility hidden
image

There a display none
image

And there, the one prefer: display none and no min-witdh: 100px
image

File: src/frontend/js/widgets/CourseProductItem/components/PaymentButton/index.tsx (hunk 6 above)
Comment (rlecellier) [reply src/frontend/js/widgets/CourseProductItem/components/PaymentButton/index.tsx:227]: I didn't find any token for button's `font-size` [in cunningham](https://github.com/openfun/cunningham/blob/main/packages/react/src/cunningham-tokens.css#L198-L204).

Also, i realize that our [tokens.css in richie](https://github.com/openfun/richie/blob/master/src/frontend/scss/vendors/css/cunningham-tokens.css) have fewer entries (118L in richie vs 204L in cunningham)

File: src/frontend/js/components/RegisteredAddress/index.tsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (jbpenrath) [reply src/frontend/js/components/RegisteredAddress/index.tsx:107]: I do not understand your point. Setting visibility to hidden should not impact layout

File: src/frontend/js/widgets/CourseProductItem/components/PaymentButton/index.tsx (hunk 6 above)
Comment (jbpenrath) [reply src/frontend/js/widgets/CourseProductItem/components/PaymentButton/index.tsx:227]: Where did you see a token file of 204L in Cunningham ?

Comment (jbpenrath) [reply src/frontend/js/widgets/CourseProductItem/components/PaymentButton/index.tsx:227]: I realized that button component within Cunningham has no `font-family` set. I think that's why in Richie, some buttons are using Montserrat and other hind...

**Review by jbpenrath — APPROVED** [review]
LGTM ✅

**Pull request merged**